frontend/src/components/file/AdvancedFileUpload.vue

```diff
@@ -23,17 +23,37 @@ const {
   disabled?: boolean;
 }>();
 
+const emit = defineEmits<{
+  (event: 'activityGenerated', activityId: string, submissionId: string): void;
+}>();
+
 // Store
 const { getConfig } = storeToRefs(useConfigStore());
 const submissionStore = useSubmissionStore();
 
 // State
+const curActivityId: Ref<string | undefined> = ref(undefined);
 const fileInput: Ref<any> = ref(null);
 const uploading: Ref<Boolean> = ref(false);
 
 // Actions
 const toast = useToast();
 
+async function generateActivityId() {
+  try {
+    const response = await submissionService.updateDraft({});
+    if (response.data?.activityId && response.data?.submissionId) {
+      emit('activityGenerated', response.data.activityId, response.data.submissionId);
+      return response.data.activityId;
+    } else {
+      return undefined;
+    }
+  } catch (error) {
+    toast.error('Failed to generate activity ID');
+    return undefined;
+  }
+}
+
```

Comment:
This is a lot of logic unrelated to file uploading. Can this be moved to the intake form?

Reply:
done